xla/service/gpu/gemm_fusion_autotuner.cc

```diff
@@ -1165,11 +1166,60 @@ absl::StatusOr<bool> GemmFusionAutotuner::Run(
                                             ? "(with correctness check)"
                                             : "(without correctness check)";
 
-    VLOG(1) << "Autotuning " << gemm_config_sets.size() << " fusions "
-            << correctness_check_str << ".";
+    const bool shard_autotuning = debug_options.xla_gpu_shard_autotuning() &&
+                                  config_.shard_count() > 1 &&
+                                  total_fusion_count > 0;
+    // Trim the set of configs to what this rank has to run.
+    if (shard_autotuning) {
+      if (config_.key_value_store() == nullptr) {
+        return absl::FailedPreconditionError(
+            "Sharded autotuning requested but key-value store is missing.");
+      }
+      const int bucket_size = (total_fusion_count + config_.shard_count() - 1) /
+                              config_.shard_count();
+      const int start = bucket_size * config_.shard_index();
+      const int end = std::min(start + bucket_size, total_fusion_count);
+      if (start >= end) {
+        gemm_config_sets.clear();
```

Comment:
Not entirely sure when do we hit this case.

Reply:
2 fusions, 5 processes -> bucket_size = 1; for process 5: start = 5, end = 2